src/main/1.3/scala/com/holdenkarau/spark/testing/StreamingSuiteCommon.scala

```diff
@@ -182,11 +180,15 @@ private[holdenkarau] trait StreamingSuiteCommon extends Logging with SparkContex
    * Returns a sequence of items for each RDD.
    */
   private[holdenkarau] def runStreams[V: ClassTag](
-      outputStream: TestOutputStream[V],
-      ssc: TestStreamingContext,
-      numBatches: Int,
-      numExpectedOutput: Int
-    ): Seq[Seq[V]] = {
+     outputStream: TestOutputStream[V],
+     ssc: TestStreamingContext,
+     numBatches: Int,
+     expectedOutput: Seq[Seq[V]],
+     verifyOutputTillNow: (mutable.Queue[Seq[V]], mutable.Queue[Seq[V]]) => Unit
```

Comment:
We are really collecting the entire thing, but we compare instantly. This means that may be before collecting the second DStream, the first Dstream could be matched then with the expected output.